### DISCUSSION TITLE
Map format

### DISCUSSION THREAD

**Discussion by Skoll444** [rinigus/osmscout-server#443] [General]
Good evening!
I have some questions.
As I understand it, the server uses maps from the site openmaptiles.org to work offline. And the format of the displayed map has .sqlite. 
Can I use maps such as the ".obf" formats that OsmAnd creates or the ".osm" format maps from OpenStreetMap?
Can I use any satellite maps to work offline?
Is it possible to use the server to display different types of maps? And if so, how does it work?
Thank you for your time.
Sincerely, Skoll

**Comment by rinigus** [comment]
Internally, it is using 3 different formats for different aspects of the map function:

- MBTiles for map display
- Combination of SQLite and external indexes for search
- Valhalla tiles for routing

All those formats are using different scripts to import the data from OSM. Code: https://github.com/rinigus/osmscout-server/tree/master/scripts/import

> Can I use maps such as the ".obf" formats that OsmAnd creates or the ".osm" format maps from OpenStreetMap?

You can after you convert it to those 3 formats. Would require some effort though. Note that geocoding is using Nominatim database for intermediate format

> Can I use any satellite maps to work offline?

Not as it is now. You would have to write corresponding functionality and get a good source for satellite tiles

> Is it possible to use the server to display different types of maps? And if so, how does it work?

If you mean whether it is possible to apply different styling: yes, but styles have to be written. See Mapbox or MapLibre styling documentation. We currently use OpenMapTiles schema for MBTiles

If you need some extra data, that has to be imported as well.

If you want to invest time into it, I can try to point towards information and help if you get stuck. So far, unfortunately, nobody really was ready to invest their time to make their wishes a reality :) . So, I am giving the pointers above, but not more specific info as a field is relatively broad and it makes sense to start with some specific question/problem.
